src/workerd/server/server.c++

```diff
@@ -2538,8 +2538,12 @@ kj::Promise<void> Server::listenOnSockets(config::Config::Reader config,
     auto rewriter = kj::heap<HttpRewriter>(httpOptions, headerTableBuilder);
 
     tasks.add(listener
-        .then([this, &service, rewriter = kj::mv(rewriter), physicalProtocol]
+        .then([this, &service, rewriter = kj::mv(rewriter), physicalProtocol, name]
               (kj::Own<kj::ConnectionReceiver> listener) mutable {
+      KJ_IF_MAYBE(stream, controlOverride) {
+        auto message = kj::str("{\"event\":\"listen\",\"socket\":\"", name, "\",\"port\":", listener->getPort(), "}\n");
+        (*stream)->write(message.begin(), message.size());
```

Comment:
What happens in the case the Node.js process closes the pipe or exits before this is written? Unlikely case, of course, but let's make sure it's tested.

Reply:
It'll throw an exception. I think that's probably fine.